src/main/java/net/mcreator/workspace/misc/WorkspaceInfo.java

```diff
@@ -178,6 +179,17 @@ public boolean hasBrewingRecipes() {
 		return false;
 	}
 
+	public boolean hasGameRulesOfType(String type) {
+		for (ModElement element : workspace.getModElements())
+			if (element.getType() == ModElementType.GAMERULE) {
+				GeneratableElement ge = element.getGeneratableElement();
+				if (ge instanceof GameRule)
+					if (((GameRule) ge).type.equals(type))

```

Comment:
As we are now using J16, you can simplify the code with 
```
[...]
if (element.getGeneratableElement() instanceof GameRule gameRule)
    if (gameRule.type.equals(type))
[...]
```

Reply:
:heavy_check_mark: Fixed

Reply:
P. S. Why didn't that happen to recipes then?